code/modules/mob/mob_misc_procs.dm

```diff
@@ -158,16 +158,26 @@
 		return U.sensor_mode
 	return SUIT_SENSOR_OFF
 
-/proc/offer_control(mob/M)
-	to_chat(M, "Control of your mob has been offered to dead players.")
+/proc/offer_control(mob/M, hours, hide_role)
+	if(HAS_TRAIT(M, TRAIT_BEING_OFFERED))
+		return
+	var/minhours
+	ADD_TRAIT(M, TRAIT_BEING_OFFERED, "build_mode")
```

Comment:
Not necessarily, could you make it be `admin_offer` or something?

Reply:
do you mean name of the trait or the source of the trait? If it's the latter, its a suggestion from coding chat

Reply:
Source of the trait